ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -848,43 +848,54 @@ translateLedgerStateShelleyToAllegraWrapper ::
 translateLedgerStateShelleyToAllegraWrapper =
     ignoringBoth $
       TranslateLedgerState {
-          translateLedgerStateWith = \_epochNo ->
-                -- This 'withLedgerTables' creates out of thin air the
-                -- differences that spawn from the Shelley->Allegra transition,
-                -- using the hard-coded deadCardanoAVVMs.
-                --
-                -- In the near future, we expect the ledger to keep track of
-                -- these and expose a function so that we can get them from
-                -- inside the ledger state we are translating.
-                --
-                -- > - $ Map.map (unTxOutWrapper . SL.translateEra' () . TxOutWrapper)
-                -- > - $ SL.unUTxO deadCardanoAVVM
-                -- > + $ SL.getRedeemersToDelete $ shelleyLedgerState ls
-                --
-                -- In the long run, the ledger will already use ledger states
-                -- parametrized by the map kind and therefore will already
-                -- provide the differences in this translation.
-                flip withLedgerTables (ShelleyLedgerTables
-                                       $ ApplyDiffMK
-                                       $ HD.UtxoDiff
-                                       $ Map.map (`HD.UtxoEntryDiff` HD.UedsDel)
-                                       $ Map.map (unTxOutWrapper . SL.translateEra' () . TxOutWrapper)
-                                       $ SL.unUTxO deadCardanoAVVM
-                                      )
-              . unFlip
-              . unComp
-              . SL.translateEra' ()
-              . Comp
-              . Flip
-              -- This 'stowLedgerTables' + 'withLedgerTables' injects the values
-              -- from the hard-coded deadCardanoAVVMs so that the ledger, which
-              -- expects an 'l EmptyMK' actually has those entries in the UTxO
-              -- and destroys them, modifying the reserves accordingly.
+          translateLedgerStateWith = \_epochNo ls ->
+              -- In the Shelley to Allegra transition, the AVVM addresses have
+              -- to be deleted, and their balance has to be moved to the
+              -- reserves. For this matter, the Ledger keeps track of these
+              -- small set of entries since the Byron to Shelley transition and
+              -- provides them to us through 'shelleyToAllegraAVVMsToDelete'.
               --
-              -- Once the ledger keeps track of these values, this operation
-              -- will no longer be needed.
-              . stowLedgerTables
-              . flip withLedgerTables (ShelleyLedgerTables $ ApplyValuesMK $ HD.UtxoValues $ SL.unUTxO deadCardanoAVVM)
+              -- In the long run, the ledger will already use ledger states
+              -- parametrized by the map kind and therefore will already provide
+              -- the differences in this translation.
+              let avvms           = SL.unUTxO $ fromSMaybe
```

Comment:
FYI, one of my requests on Clarke's PR was to use a type family which would avoid the need for this `error` case here.